dotCMS/src/main/java/com/dotcms/util/content/json/PopulateContentletAsJSONUtil.java

```diff
@@ -0,0 +1,398 @@
+                                        contentletList.stream()
+                                                .map(this::toJSON)
+                                                .collect(Collectors.toList())
+                                )// Transform the contentlets into a list of json strings
+                                .orElse(Collections.emptyList());
+
+                        for (var jsonData : jsonDataArray) {
+                            // Write the json representation of the contentlet into the file
+                            fileWriter.write(jsonData);
+                            fileWriter.newLine();
+                        }
+
+                        Logger.debug(this, String.format("Added [%s] records for update to temp file", jsonDataArray.size()));
+
+                    } else {
+                        hasRows = false;
+                    }
+                }
+
+            } while (hasRows);
+
+            // Flush the writer to the file
+            fileWriter.flush();
+
+            // Close the cursor
+            stmt.execute(CLOSE_CURSOR);
+            if (DbConnectionFactory.isMsSql()) {
+                stmt.execute(DEALLOCATE_CURSOR_MSSQL);
+            }
+        } finally {
+            CloseUtils.closeQuietly(fileWriter);
+        }
+
+        Logger.info(this, "-- Records found to process: " + recordsProcessed);
+    }
+
+    /**
+     * This method processes a file that contains all the contentlets that need to be updated with the contentlet_as_json
+     *
+     * @param taskDataFile
+     * @throws IOException
+     */
+    @WrapInTransaction
+    private void processFile(final File taskDataFile) throws IOException {
+
+        Logger.info(this, "Updating records with missing Contentlet as JSON");
+
+        final MutableInt totalUpdateAffected = new MutableInt(0);
+
+        try (final Stream<String> streamLines = Files.lines(taskDataFile.toPath())) {
+
+            streamLines.forEachOrdered(line -> this.processLine(line, totalUpdateAffected));
+
+            if (!this.paramsUpdate.isEmpty()) {
+                this.doUpdateBatch(totalUpdateAffected);
+            }
+        } finally {
+            Logger.info(this, "-- total updates: " + totalUpdateAffected.intValue());
+        }
+
+        Logger.info(this, "Updated records with missing Contentlet as JSON");
+    }
+
+    /**
+     * Processes the given line preparing the params for the batch update.
+     *
+     * @param line Line with the json representation of the contentlet.
+     * @return
+     * @throws JsonProcessingException
+     */
+    private void processLine(final String line, final MutableInt totalInsertAffected) {
+
+        try {
+            var contentlet = ContentletJsonHelper.INSTANCE.get().immutableFromJson(line);
+
+            final Object contentletAsJSON;
+            if (DbConnectionFactory.isPostgres()) {
+                PGobject jsonObject = new PGobject();
```

Comment:
It would be great to have a DotPageObject that can be used such as 

contentAsJson = new DotPGObject (json);
so you can resume to

final Object contentletAsJSON = DbConnectionFactory.isPostgres()? new DotPGObject (json): line;